providers/redfish/firmware.go

```diff
@@ -124,6 +140,13 @@ func (c *Conn) FirmwareInstall(ctx context.Context, component, applyAt string, f
 		taskID = strings.Split(resp.Header.Get("Location"), "JID_")[1]
 	}
 
```

Comment:
lets move both these lookups for the taskID into its own function

Reply:
yes, done.